hr_holidays_public/models/hr_holidays_public.py

```diff
@@ -53,6 +53,14 @@ def name_get(self):
             result.append((rec.id, rec.display_name))
         return result
 
+    @api.model
+    @api.returns("hr.holidays.public.line")
+    def get_holidays_list_by_dates(self, start_dt, end_dt, employee_id=None):
+        holidays_lines = self.env["hr.holidays.public.line"]
```

Comment:
My idea was the contrary: the base method to be the date range method, and this is not correct, as your adding more lines than expected.